tests_common/test_utils/db.py

```diff
@@ -73,7 +73,12 @@ def initial_db_init():
     from airflow.www.extensions.init_appbuilder import init_appbuilder
     from airflow.www.extensions.init_auth_manager import get_auth_manager
 
+    from tests_common.test_utils.compat import AIRFLOW_V_3_0_PLUS
+
     db.resetdb()
+    if AIRFLOW_V_3_0_PLUS:
+        db.downgrade(to_revision="5f2621c13b39")
+        db.upgradedb(to_revision="head")
```

Comment:
Do we need to do this in every single test run, or could we limit this to the test job specifically checking the db upgrade/downgrade flow?

Reply:
I can’t see how to limit it to only the test that runs migration. My thinking is that we can’t limit it due to running tests in parallel and that doing it specifically for the /utils/db tests could affect other tests running. 

Also, the code only runs when we have —with-db-init flag on tests, which i think happens when tests are initialized and not on every one of the tests

Reply:
Yeah. One tim only per session (and only when DB is not initialized or when `--with-db-init`)  - so optimising it is not really needed.

Reply:
BTW. We should indeed optimize some of the fixtures I think - but I'd say we should measure the impact and optimise those that are really slowing things down: see https://github.com/apache/airflow/issues/43488 -> otherwise `premature optimisation is the root of all evil (and complexity in the code)`